File: levelledmobs-plugin-bukkit/src/main/java/me/lokka30/levelledmobs/bukkit/util/StringUtils.java
package me.lokka30.levelledmobs.bukkit.util;

import java.util.function.Supplier;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class StringUtils {

    private StringUtils() {}

    @NotNull
    public static String replaceIfExists(
        final @NotNull String str,
        final @NotNull String target,
        final @NotNull Supplier<@NotNull String> operation
    ) {
        if(str.contains(target)) {
            return str.replace(target, operation.get());
        }
        return str;
    }

    @NotNull
    public static String emptyIfNull(
        final @Nullable String str
    ) {
        return str == null ? " " : str;
    }

}

File: levelledmobs-plugin-bukkit/src/main/kotlin/io/github/arcaneplugins/levelledmobs/bukkit/util/StringUtils.kt
package io.github.arcaneplugins.levelledmobs.bukkit.util

import java.util.function.Supplier

object StringUtils {
    fun replaceIfExists(
        str: String,
        target: String,
        operation: Supplier<Any>
    ): String {
        return if (str.contains(target)) {
            str.replace(target, operation.get().toString())
        }
        else str
    }

    fun emptyIfNull(
        str: String?
    ): String {
        return str ?: " "
    }
}